src/main/java/ai/elimu/web/SignOnControllerWeb3.java

```diff
@@ -33,7 +33,7 @@ public class SignOnControllerWeb3 {
     @Autowired
     private ContributorDao contributorDao;
 
-    @RequestMapping("/sign-on/web3")
+    @RequestMapping(value="/sign-on/web3", method=RequestMethod.GET)
```

Comment:
Looks like `@GetMapping` annotation can be used in such case, to restrict handling of other HTTP methods in future

Reply:
@vrudas Thanks for sharing. I see that there is a `@PostMapping` as well. Using `@RequestMapping` without any `method` parameter should resolve automatically to `GET`, but the MockMvc framework did not pick up on this.

Do you think we should refactor all of our controllers to use `@GetMapping`/`@PostMapping` instead of `@RequestMapping`?

Reply:
@vrudas Please see https://discord.com/channels/920940233225302067/930695750088810536/997312944549400606

Reply:
@nya-elimuai Usage or `@GetMapping`/`@PostMapping` instead of `@RequestMapping` is more preferred from a security point, because we will reduce amount of possible vulnerability attacks of random curl requests for each of HTTP methods